addon/edit/matchbrackets.js

```diff
@@ -106,13 +121,15 @@
       cm.off("cursorActivity", doMatchBrackets);
     if (val) {
       cm.state.matchBrackets = typeof val == "object" ? val : {};
+      cm.state.matchBrackets.fatCursor = 
+          cm.getWrapperElement().classList.contains('cm-fat-cursor');
```

Comment:
Could you make this so that, instead of mutating this object, you check for `null` when actually running `findMatchingBracket`, and only if the user did not provide an option for `fatCursor`, you default it based on the presence of the class name.

Also, we still support old browsers that don't have `classList`, so a regexp check on `className` is preferred.